bsi_zoo/benchmark.py

```diff
@@ -91,35 +117,61 @@ def __init__(
         self.random_state = random_state
         self.memory = memory if isinstance(memory, Memory) else Memory(memory)
         self.n_jobs = n_jobs
+        self.do_spatial_cv = do_spatial_cv
+        self.estimator_extra_params = estimator_extra_params
 
     def run(self, nruns=2):
         rng = check_random_state(self.random_state)
         seeds = rng.randint(low=0, high=2 ** 32, size=nruns)
 
         estimator = self.memory.cache(self.estimator)
 
-        results = Parallel(n_jobs=self.n_jobs)(
-            delayed(_run_estimator)(
-                self.subject,
-                estimator,
-                self.metrics,
-                this_data_args,
-                this_estimator_args,
-                seed,
-                estimator_name=self.estimator.__name__,
-                memory=self.memory,
+        if do_spatial_cv:
+            # dont make param grid for estimator args
+            results = Parallel(n_jobs=self.n_jobs)(
+                delayed(_run_estimator)(
+                    self.subject,
+                    estimator,
+                    self.metrics,
+                    this_data_args,
+                    self.estimator_args,
+                    seed,
+                    estimator_name=self.estimator.__name__,
+                    memory=self.memory,
+                    do_spatial_cv=self.do_spatial_cv,
+                    extra_params=estimator_extra_params
+                )
+                for this_data_args, seed in itertools.product(
+                    ParameterGrid(self.data_args), seeds
+                )
             )
-            for this_data_args, seed, this_estimator_args in itertools.product(
-                ParameterGrid(self.data_args), seeds, ParameterGrid(self.estimator_args)
+        else:
+            results = Parallel(n_jobs=self.n_jobs)(
+                delayed(_run_estimator)(
+                    self.subject,
+                    estimator,
+                    self.metrics,
+                    this_data_args,
+                    this_estimator_args,
+                    seed,
+                    estimator_name=self.estimator.__name__,
+                    memory=self.memory,
+                    do_spatial_cv=self.do_spatial_cv,
+                )
+                for this_data_args, seed, this_estimator_args in itertools.product(
+                    ParameterGrid(self.data_args),
+                    seeds,
+                    ParameterGrid(self.estimator_args),
+                )
```

Comment:
@anujanegi this is quiet some code duplication. To me you should have a solver API and then the benchmark class should not worry if it's SpatialCVSolver or a plain Solver. do you see where I am heading?